portfolio/src/main/webapp/calendarApi.js

```diff
@@ -0,0 +1,298 @@
+      });
+}
+
+/** Signs user in if not logged in, and signs user out otherwise. */
+function handleAuth() {
+  if (GoogleAuth.isSignedIn.get()) {
+    GoogleAuth.signOut();
+  } else {
+    // User is not signed in. Start Google auth flow.
+    GoogleAuth.signIn().catch(function(error) {
+      handleImportAuthError(error);
+    });
+  }
+}
+
+/**
+ * Updates import message box based on the error during authentication process
+ * for importing
+ */
+function handleImportAuthError(e) {
+  $('#import-calendar-message').removeClass('d-none');
+  if (e.error === ERROR_CODE_POPUP_CLOSED) {
+    $('#import-calendar-message')
+        .text(
+            'It seems like you didn\'t complete the authorization process. ' +
+            'Please click the Login button again.');
+  } else if (e.error === ERROR_CODE_ACCESS_DENIED) {
+    $('#import-calendar-message')
+        .text(
+            'You didn\'t give permission to view your Google Calendar, ' +
+            'so your calendar events cannot be viewed or imported.');
+  }
+  $('#import-calendar-message').show();
+}
+
+/** Disconnects current user authentication. */
+function revokeAccess() {
+  GoogleAuth.disconnect();
+}
+
+/** Updates the calendar view and button visibility based on login status. */
+function updateCalendarView() {
+  var user = GoogleAuth.currentUser.get();
+  var isAuthorized = user.hasGrantedScopes(SCOPE_READ_ONLY);
+  if (isAuthorized) {
+    showCalendarView(user);
+    $('#calendar-auth-button').attr('disabled', 'true');
+    $('#calendar-auth-button').attr('aria-disabled', 'true');
+    $('#import-calendar-button').removeClass('d-none');
+    $('#import-calendar-button').show();
+  } else {
+    $('#calendar-auth-button').attr('disabled', 'false');
+    $('#calendar-auth-button').attr('aria-disabled', 'false');
+    $('#import-calendar-button').hide();
+  }
+}
+
+/** Retrives the date that the user has picked for the scheduling. */
+function getUserPickedDate() {
+  const userPickedDate = $('#date-picker').val().split('-');
+  const year = userPickedDate[0];
+  const month = userPickedDate[1];
+  const date = userPickedDate[2];
+
+  const pickedDate = new Date();
+  pickedDate.setFullYear(year);
+  pickedDate.setMonth(month - 1);  // month is zero-indexed.
+  pickedDate.setDate(date);
+
+  return pickedDate;
+}
+
+/**
+ * Print the summary and start datetime/date of the events for the
+ * day that user has picked in the nav bar.
+ * If no events are found a message is displayed on the UI.
+ */
+function listUpcomingEvents() {
+  const timeRangeStart = getUserPickedDate();
+  const timeRangeEnd = new Date();
+  timeRangeEnd.setFullYear(timeRangeStart.getFullYear());
+  timeRangeEnd.setMonth(timeRangeStart.getMonth());
+  timeRangeEnd.setDate(timeRangeStart.getDate() + 1);
+
+  // Retrieves events on the user's calendar for the day
+  // that the user has picked in the nav bar. 
+  gapi.client.calendar.events
+      .list({
+        'calendarId': 'primary',
```

Comment:
We don't need the single quotes in the keys.

Reply:
Updated.